### PR TITLE
Initial import of the Vault role

### PR DESCRIPTION
**Pull request by dasJ**
This role is able to set up a HashiCorp Vault.
See the README.md file for more information.

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+tls_cert: /etc/ssl/vault.pem
+tls_key: /etc/ssl/vault.key
+tls_min_version: tls12
+tls_prefer_server_suites: true
+# Backend
+backend: file
+backend_config:
+  path: "{{ db_path | mandatory }}"
+# Leases
+max_lease_ttl: 768h
```
Comment (Scriptkiddi) [inline defaults/main.yml]: this is not a useful lease time please adjust to something resonable

File: defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+default_lease_ttl: 768h
```
Comment (Scriptkiddi) [inline defaults/main.yml]: this is not a useful lease time please adjust to something resonable

File: tasks/build.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
+---
+- name: Install dependencies (apt)
+  apt:
```
Comment (Scriptkiddi) [inline tasks/build.yml]: use the package command instead of apt or pacman

File: tasks/build.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  register: git
+
+- name: Bootstrap Vault
+  command:
+    cmd: "make bootstrap"
+    chdir: "{{ ansible_env.HOME }}/go/src/github.com/hashicorp/vault"
+  environment:
+    GOPATH: "{{ ansible_env.HOME }}/go"
+    PATH: "{{ ansible_env.HOME}}/go/bin:{{ansible_env.PATH }}"
+  register: out
```
Comment (Scriptkiddi) [inline tasks/build.yml]: out is a non descriptive var name please change

File: tasks/build.yml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  # Never changed because Makefile will always do something
+  changed_when: false
+  when: git.changed
+
+- name: Fetch go variables
+  shell: "echo -n \"$(go env GOOS)_$(go env GOARCH)\""
+  register: goout
+  changed_when: false
+
+# Workaround because the copy module is not capable of hashing
```
Comment (Scriptkiddi) [inline tasks/build.yml]: this seems wrong have you looked at filters https://docs.ansible.com/ansible/playbooks_filters.html#hashing-filters

File: templates/vault.hcl.j2 [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+{% endif %}
+
+# TCP Listener
+listener "tcp" {
+	address = "{{ listen }}:{{ port }}"
+{% if cluster_address is defined %}
+	cluster_address = "{{ cluster_address }}"
+{% endif %}
+	tls_disable = {{ disable_tls | int }}
+{% if tls_disable is defined and not tls_disable %}
```
Comment (Scriptkiddi) [inline templates/vault.hcl.j2]: you set tls_disable in your default/main.yml so it should always be defined depending on how we want go about this its useless code.
In my opinion its the right way to do things

File: tasks/build.yml (hunk 5 above)
Comment (dasJ) [inline tasks/build.yml]: Yes, the filter hashes strings, but I want to hash files

File: tasks/build.yml (hunk 4 above)
Comment (dasJ) [inline tasks/build.yml]: Why? This variable is locally scoped and will not be used after this task

File: defaults/main.yml (hunk 2 above)
Comment (dasJ) [inline defaults/main.yml]: It's the default

File: tasks/build.yml (hunk 3 above)
Comment (dasJ) [inline tasks/build.yml]: No

**Comment by dasJ** [comment]
@Scriptkiddi Please re-review

**Pull request merged**